Added explanation of indexOf method in "Useful string methods" page (#14350)
mdn/content: files/en-us/learn/javascript/first_steps/useful_string_methods/index.md

}
```

## Finding the position of a substring in a string

> You can search for and find the position of a substring inside a larger string using the {{jsxref("String.prototype.indexOf()", "indexOf()")}} method. This method takes two {{glossary("parameters")}} – the substring that you want to search for, and an optional parameter that specifies the starting point of the search.

Reviewer: ```suggestion
You can find the position of a substring inside a larger string using the {{jsxref("String.prototype.indexOf()", "indexOf()")}} method. This method takes two {{glossary("parameter", "parameters")}} – the substring that you want to search for, and an optional parameter that specifies the starting point of the search.
```

> If the string contains the substring, it returns the index of the first occurrence of the substring, and `-1` otherwise.

Reviewer: ```suggestion
If the string contains the substring, `indexOf()` returns the index of the first occurrence of the substring. If the string does not contain the substring,  `indexOf()` returns `-1`.
```

Author: @wbamberg Thanks for the feedback.

I'm not sure whether the suggested change is consistent with the rest of the content on the page. For instance, there is a line under the ["Testing if a string contains a substring"](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/First_steps/Useful_string_methods#testing_if_a_string_contains_a_substring) section that says:

"It returns true if the string contains the substring, and false otherwise."

Reviewer: As a rule I think "It" is a bad word in technical documentation because you always have to think about what it refers to. It's worse the further it is from the original thing, but for clarity IMO it's usually clearer to repeat the name of the thing that to use "it", even when it seems a bit clumsy to do so. In the case you linked of `includes()` I would also prefer to use the name of the method, like:

"The `includes()` method returns true if the string contains the substring, and false otherwise."

As for splitting the sentence into two: the sentence here is longer and has a more complex structure than the one for `includes()`. In particular there's significant distance between "otherwise" and the thing it is referring to "If the string contains the substring" that (IMO) it is clearer to have two sentences.

Author: Thanks for clearing that up, @wbamberg

```js

> let tagline = "MDN - Resources for developers, by developers";

Reviewer: ```suggestion
const tagline = 'MDN - Resources for developers, by developers';
```

> console.log(tagline.indexOf("developers")); // 20

Reviewer: ```suggestion
console.log(tagline.indexOf('developers')); // 20
```

```

Starting at `0`, if you count the number of characters (including the whitespace) from the beginning of the string, the first occurrence of the substring `"developers"` is at index `20`.

```js

> console.log(tagline.indexOf("x")); // -1

Reviewer: ```suggestion
console.log(tagline.indexOf('x')); // -1
```

```

This, on the other hand, returns `-1` because the character `x` is not present in the string.

> So now that you know how to find the first occurrence of a substring, how do you go about finding the subsequent occurrences? You can do that by passing in a value that's greater than the index of the previous occurrence as the second parameter to the method.

Reviewer: ```suggestion
So now that you know how to find the first occurrence of a substring, how do you go about finding subsequent occurrences? You can do that by passing in a value that's greater than the index of the previous occurrence as the second parameter to the method.
```

```js

> let firstOccurence = tagline.indexOf("developers");
> let secondOccurence = tagline.indexOf("developers", firstOccurence + 1);
> 
> console.log(firstOccurence); // 20
> console.log(secondOccurence); // 35

Reviewer: ```suggestion
const firstOccurrence = tagline.indexOf('developers');
const secondOccurrence = tagline.indexOf('developers', firstOccurrence + 1);

console.log(firstOccurrence); // 20
console.log(secondOccurrence); // 35
```
